model.js

```diff
@@ -60,5 +60,20 @@ const fetchSingleArticle = (id) => {
         }
     })    
 };
-module.exports = { fetchAllTopics, fetchAllArticles, fetchSingleArticle, fetchArticleComments };
 
+const sendComment = (id, comment) => {
+  const sqlQuery = `
+INSERT INTO comments (article_id, author, body)
+VALUES ($1, (SELECT username FROM users WHERE username = $2), $3)
+RETURNING *;`
+  return db
+  .query(sqlQuery, [id, comment.username, comment.body])
+  .then((result) => {
+    if (result.rows.length === 0){
+      return Promise.reject({ status: 404, msg: "Not found!" })
+    } else {
+    return result.rows[0]
+    }
+  })
+}
```

Comment:
you'll need to sort error handling for if the wrong id is used etc (maybe you could consider doing this in the controller function - it's up to you but you need to make sure it's there).

Reply:
I'm not sure what you mean as I thought I was handling the errors the controller already?
I have the catch block at the end of the postComment function so I thought that this is catching the error and returning it to the app? In the model, if the id is invalid, the sql query doesn't return anything and so if the array is empty, it responds with the "not found" error...

Reply:
I just can't read apparently 😬 Go for it!!!